website/docs/reference/global-configs.md

```diff
@@ -103,6 +103,41 @@ config:
 
 </File>
 
+<VersionBlock firstVersion="1.1">
+
+### Cache database objects for selected resource
+
+:::caution Experimental config flag
+This should be considered an _unstable_ config flag because it is experimental and subject to change. We reserve the right to make breaking changes to this config flag.
+:::
+
+When true, dbt caches database objects relevant to selected resource. Otherwise, dbt will cache all database objects related to the project.
+
+For example, to improve speed and performance while focused on Salesforce models you would select the source and use the `cache-selected-only` flag:
+
+```text
+
+$ dbt build --select source:salesforce 
+$ dbt --cache-selected-only run
+
+```
```

Comment:
Meta-comment: dbt doesn't actually cache sources, only buildable resources (models, seeds, snapshots, tests if `--store-failures`). Again, we're missing broader explanatory resources about caching, which we can only make up for so much in this one little section.
```suggestion
For example, if your project materializes Salesforce models in a dedicated schema, and you want to focus on building those models, you can improve start-up speed by using the `--cache-selected-only` flag:

```text
$ dbt --cache-selected-only build --select salesforce
```
This will run fewer metadata queries upfront, and start building faster.

Reply:
Thanks for adding the context, @jtcohen6 I think adding caching content to the docs and linking to it will really help!